pkg/cache/context.go

```diff
@@ -499,10 +499,31 @@ func (ctx *Context) GetApplication(appID string) interfaces.ManagedApp {
 func (ctx *Context) RemoveApplication(appID string) error {
 	ctx.lock.Lock()
 	defer ctx.lock.Unlock()
-	if _, exist := ctx.applications[appID]; exist {
+	if app, exist := ctx.applications[appID]; exist {
+		var terminatedStates = events.States().Task.Terminated
+		terminatedNum := 0
+		taskmessage := "Task: "
+		for _, task := range app.taskMap {
+			for _, states := range terminatedStates {
+				if states == task.GetTaskState() {
+					terminatedNum++
+				}
+			}
+			taskmessage = taskmessage + task.alias + ", "
```

Comment:
this will leave a "," in the end of the string. This can be fixed by using https://golang.org/pkg/strings/#Join
another thing is we can simplify the code in `context.go` by introducing a function in `application.go`, like the following

```
// returns a array of task alias that in non-terminated state
nonTerminatedTaskAlias:= app.getNonTerminatedTaskAlias()
if len(nonTerminatedTasks) > 0 {
  return fmt.Errorf("failed to remove application %s because it still has tasks in non-terminated state, tasks:  %s", appID, strings.Join(nonTerminatedTaskAlias, ","))
}
 
```

Reply:
if this makes sense to you, pls add UT for `app.getNonTerminatedTaskAlias()` too. So we can cover cases like:
- app doesn't have any task
- app has some tasks, all in the terminated state
- app has some tasks, all in non-terminated state
- app has some tasks that in both terminated or non-terminated state